src/cdpy/df.py

```diff
@@ -7,48 +7,69 @@ class CdpyDf(CdpSdkBase):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-    def list_environments(self, only_enabled=False, name=None):
+    def list_services(self, only_enabled=False, env_crn=None, df_crn=None, name=None):
         result = self.sdk.call(
-            svc='df', func='list_environments', ret_field='environments', squelch=[
+            svc='df', func='list_services', ret_field='services', squelch=[
                 Squelch(value='NOT_FOUND', default=list(),
-                        warning='No Environments for DataFlow found in Tenant')
+                        warning='No DataFlow Deployments found')
             ],
             pageSize=self.sdk.DEFAULT_PAGE_SIZE
         )
         if only_enabled:
             result = [x for x in result if x['status']['state'] not in ['NOT_ENABLED']]
         if name is not None:
             result = [x for x in result if x['name'] == name]
+        if df_crn is not None:
+            result = [x for x in result if x['crn'] == df_crn]
+        if env_crn is not None:
+            result = [x for x in result if x['environmentCrn'] == env_crn]
         return result
 
-    def describe_environment(self, env_crn: str = None):
+    def describe_service(self, df_crn: str = None, env_crn: str = None):
+        if df_crn is not None:
+            resolved_df_crn = df_crn
+        elif env_crn is not None:
+            services = self.list_services(env_crn=env_crn)
+            if len(services) == 0:
+                return None
+            elif len(services) == 1:
+                resolved_df_crn = services[0]['crn']
+            else:
+                self.sdk.throw_error(
+                    CdpError('More than one DataFlow service found for env_crn, please try list instead')
```

Comment:
Why do we need this? DF supports more than one service per env, no?

Reply:
yeah it's a sanity check so the user is less surprised than I was when they found it wasn't a strict 1:1 mapping of env to df service